core/src/col_metadata.rs

```diff
@@ -44,47 +47,59 @@ pub struct MongoColMetadata {
 }
 
 impl MongoColMetadata {
-    pub fn new(
+    pub fn new_metadata_from_bson_type_info(
         _current_db: &str,
         datasource_name: String,
         field_name: String,
-        field_schema: Schema,
+        bson_type_info: BsonTypeInfo,
         nullability: Nullability,
     ) -> MongoColMetadata {
-        let bson_type_info: BsonTypeInfo = field_schema.into();
-
         MongoColMetadata {
             // For base_col_name, base_table_name, and catalog_name, we do
             // not have this information in sqlGetResultSchema, so these will
             // always be empty string for now.
             base_col_name: "".to_string(),
             base_table_name: "".to_string(),
+            case_sensitive: bson_type_info.is_case_sensitive,
             catalog_name: "".to_string(),
             display_size: bson_type_info.fixed_bytes_length,
-            fixed_prec_scale: false,
+            fixed_prec_scale: bson_type_info.fixed_prec_scale,
             label: field_name.clone(),
             length: bson_type_info.fixed_bytes_length,
+            literal_prefix: bson_type_info.literal_prefix,
+            literal_suffix: bson_type_info.literal_suffix,
             col_name: field_name,
             nullability,
             octet_length: bson_type_info.octet_length,
             precision: bson_type_info.precision,
             scale: bson_type_info.scale,
-            is_searchable: bson_type_info.searchable,
+            searchable: bson_type_info.searchable,
             table_name: datasource_name,
             type_name: bson_type_info.type_name.to_string(),
             sql_type: bson_type_info.sql_type,
-            non_concise_type: match bson_type_info.sql_type {
-                SqlDataType::TIMESTAMP => SqlDataType::DATETIME,
-                x => x,
-            },
-            sql_code: match bson_type_info.sql_type {
-                SqlDataType::TIMESTAMP => Some(3),
-                _ => None,
-            },
-            is_unsigned: false,
+            non_concise_type: bson_type_info.non_concise_type,
+            sql_code: bson_type_info.sql_code,
+            is_unsigned: bson_type_info.is_unsigned.unwrap_or(true),
             is_updatable: false,
         }
     }
+
+    pub fn new(
+        _current_db: &str,
```

Comment:
```suggestion
        current_db: &str,
```
another place where it looks like we're using an argument so we can remove the `_`

Reply:
👍  can do